datadog/resource_datadog_monitor.go

```diff
@@ -418,6 +432,24 @@ func resourceDatadogMonitorRead(d *schema.ResourceData, meta interface{}) error
 		d.Set("query_config", queryConfig)
 	}
 
+	// The Datadog API doesn't return old timestamps or support a special value for unmuting scopes
+	// So we provide this functionality by saving values to the state
+	//If the user specified scope timestamp is -1 and its not in the API response, save it to the state
+	//If the user specified scope timestamp is in the past and its not our 0 value, remove it from the state (it won't be returned via the API)
+	//If the user specified value isn't in the API response and its not an above special case, remove it from the state so we re mute it.
+	apiSilenced := m.Options.Silenced
+	configSilenced := d.Get("silenced").(map[string]interface{})
+	for k, configuredVal := range configSilenced {
```

Comment:
So unfortunately this now breaks the functionality from [1]. Consider this scenario:

* I create a monitor with one muted scope (could be more, that doesn't matter), where the timestamp is correct future timestamp.
* I remove the scope (=> unmute it), making the `silenced` parameter empty. This will result in the API not unmuting anything, as `silenced` is not even sent by the underlying client library.
* The logic in the condition in [1] relies on the `resourceDatadogMonitorRead` to correctly read all the values from API into `d`. If this happens correctly, the condition evaluates to `true` (because `ok` is `true`, as the scope is still muted on the API side, and `!silenced` is `true` as well, because `silenced` is not set.
* Your implementation only traverses through keys of `silenced` from the current configuration (which is empty), so it won't set the silenced scope on `d` and thus [1] will fail to evaluate correctly and the scope will remain muted.

You need to traverse all keys from both current config and the API response to get this right, I think.

All the other scenarios that I could think of work fine.

[1] https://github.com/terraform-providers/terraform-provider-datadog/commit/b4d5d17b2320ee7287f9b0f88a0c6240eb5061c1